### PR TITLE
fix: should not print auth warn when adding auth to new project

### PR DESCRIPTION
**Pull request by lazpavel**
#### Description of changes
- prevents showing auth warning when adding auth to a new project

#### Description of how you validated changes
- manual and unit tests added

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/__tests__/utils/auto-updates.test.ts [hunk 1]
```diff
@@ -74,9 +79,34 @@ describe('showBuildDirChangesMessage', () => {
...
+      fsMock.existsSync.mockReturnValue(false);
+    });
+
+    it('does not call warn', async () => {
+      await showBuildDirChangesMessage();
+      expect(printerMock.warn).not.toBeCalled();
+    });
+  });
+
+  describe('project does not have auth 2', () => {
```
Comment (danielleadams) [inline packages/amplify-provider-awscloudformation/src/__tests__/utils/auto-updates.test.ts:101]: Is this named right `existsSync` returns `true` means that auth is in backend cloud right?

**Review by danielleadams — CHANGES_REQUESTED** [review]
I'm a little confused about the test naming, but other than that looks good.

File: packages/amplify-provider-awscloudformation/src/__tests__/utils/auto-updates.test.ts (hunk 1 above)
Comment (lazpavel) [reply packages/amplify-provider-awscloudformation/src/__tests__/utils/auto-updates.test.ts:101]: this is the behavior that was previously tested, folder exists but auth not in the cloud, will update test names

File: packages/amplify-cli/src/extensions/amplify-helpers/update-tracked-files.ts [hunk 2]
```diff
@@ -12,37 +12,31 @@ const {
  * Updates Cognito files that are tracked so that the diff is detected for an `amplify push`
  */
 export const updateCognitoTrackedFiles = async (): Promise<void> => {
+  if (await detectCognitoAttributesRequireVerificationBeforeUpdateDiff()) {
+    const { resourceName } = stateManager.getResourceFromMeta(stateManager.getMeta(), 'auth', 'Cognito', undefined, false)!;
+    await addExtraLineToCliInputsJson(pathManager.getBackendDirPath(), resourceName);
```
Comment (sobolk) [inline packages/amplify-cli/src/extensions/amplify-helpers/update-tracked-files.ts:17]: what does this do? (apart from adding extra line, i.e. why adding line is important?)

**Review by sobolk — APPROVED** [review]

File: packages/amplify-cli/src/extensions/amplify-helpers/update-tracked-files.ts (hunk 2 above)
Comment (lazpavel) [reply packages/amplify-cli/src/extensions/amplify-helpers/update-tracked-files.ts:17]: it enforces a resource change to be pushed

**Review by danielleadams — APPROVED** [review]

**Pull request merged**